### PR TITLE
Set string location when it's registered

### PR DESCRIPTION
**Pull request by strategio**
As the code is expanding and we have some upcoming tasks on string
registration, I finally decided to extract the code from the main class
to a new one WPML_Gutenberg_Strings_Registration (I also tried to factorize some code with the other packages but it's not easy).

https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlcore-6325

### CODE REVIEW

File: src/class-wpml-gutenberg-strings-registration.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+				$this->update_string_location( $package_data, $string );
+			}
+
+			if ( isset( $block->innerBlocks ) ) {
+				$this->register_blocks( $block->innerBlocks, $package_data );
+			}
+		}
+	}
+
+	private function update_string_location( array $package_data, stdClass $string_data ) {
```
Comment (strategio) [inline src/class-wpml-gutenberg-strings-registration.php:70]: This is the actual fix.

File: tests/phpunit/tests/test-wpml-gutenberg-string-registration.php [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+/**
+ * @group string-registration
+ */
+class Test_WPML_Gutenberg_String_Registration extends OTGS_TestCase {
+
+	/**
+	 * @test
+	 */
+	public function it_registers_strings() {
```
Comment (strategio) [inline tests/phpunit/tests/test-wpml-gutenberg-string-registration.php:11]: This test was extracted from `Test_WPML_Gutenberg_Integration`.

File: tests/phpunit/tests/test-wpml-gutenberg-string-registration.php [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+				$this->check_block_is_registered( $block, $type, $package );
+			}
+		}
+	}
+
+	/**
+	 * @test
+	 * @group wpmlcore-6325
+	 */
+	public function it_registers_strings_and_set_location() {
```
Comment (strategio) [inline tests/phpunit/tests/test-wpml-gutenberg-string-registration.php:94]: This is the test for the strings location.

**Review by BruceOnTheGo — APPROVED** [review]
@strategio This looks good. +1

**Review by andreasciamanna — APPROVED** [review]

**Pull request merged**